Use of `FE_INVALID` breaks WebAssembly builds of HDF5 1.14.5

**Describe the bug**

HDF5 1.14.5 cannot compile to WebAssembly as Emscripten does not support the `FE_*` macros.

**Expected behavior**

The compilation should succeed.

**Platform (please complete the following information)**
 - HDF5 1.14.5.
 - Ubuntu 22.04.5 LTS
 - Emscripten 3.1.68
 - Cmake
 - Multiple configure flags (see below)
 - No MPI

**Additional context**

Emscripten fails on the following chunk of code:

https://github.com/HDFGroup/hdf5/blob/6b43197b0817596f47670c6b55d26ff7f86d6bd9/src/H5Tinit_float.c#L611-L613

with:

```
# ... skipping Cmake logs...
[ 78%] Building C object src/CMakeFiles/hdf5-static.dir/H5Tinit_float.c.o
/home/luna/Code/js/scran.js/extern/hdf5/demo/hdf5-1.14.5/src/H5Tinit_float.c:612:23: error: use of undeclared identifier 'FE_INVALID'
  612 |     if (feclearexcept(FE_INVALID) != 0)
      |                       ^
1 error generated.
```

This is because many of the `FE_*` macros are not defined by Emscripten (emscripten-core/emscripten#13678). While I'm not familiar with the intricacies of the C standard, [some reading](https://en.cppreference.com/w/c/numeric/fenv/FE_exceptions) suggests that compliant implementations do not need to define these macros.

Currently I'm modifying the HDF5 source code directly so that the compilation can complete. Would you consider adding an `#ifdef FE_INVALID` around this block so that it compiles out of the box?

FWIW the MRE is:

```bash
# Unzip the downloaded tarball.
tar -xf hdf5-1.14.5.tar.gz

# Avoid problems from a random package.json in a parent directory.
echo "{}" > package.json 

# Assumes you have Emscripten installed, I'm using 3.1.68. Also
# I'm using Cmake 3.27.1, but I don't think it really matters.
emcmake cmake \
    -S hdf5-1.14.5 \
    -B build-1.14.5 \
    -DCMAKE_BUILD_TYPE=Release \
    -DCMAKE_INSTALL_PREFIX=installed \
    -DBUILD_SHARED_LIBS=OFF \
    -DBUILD_TESTING=OFF \
    -DHDF5_BUILD_EXAMPLES=OFF \
    -DHDF5_BUILD_TOOLS=OFF \
    -DHDF5_BUILD_UTILS=OFF \
    -DHDF5_BUILD_CPP_LIB=ON \
    -DHDF5_ENABLE_Z_LIB_SUPPORT=ON \
    -DZLIB_USE_EXTERNAL=OFF \
    -DHDF5_ENABLE_SZIP_SUPPORT=OFF

# Reproduces the error.
cd build-1.14.5
emmake make
```

--- a/release_docs/RELEASE.txt
+++ b/release_docs/RELEASE.txt
@@ -132,7 +132,19 @@ Bug Fixes since HDF5-1.16.0 release
 ===================================
     Library
     -------
-    -
+    - Only clear FE_INVALID when that symbol is present on the system
+
+      When we initialize the floating-point types at library startup, it's
+      possible to raise floating-point exceptions when we check which things
+      are supported. Normally, we clear these floating-point exceptions via
+      feclearexcept(FE_INVALID), but FE_INVALID may not be present on all
+      systems. Specifically, this was reported as being a problem when using
+      Emscripten 3.1.68 to compile HDF5 1.14.5 to WebAssembly.
+
+      We've added an #ifdef FE_INVALID block around the exception clearing
+      code to correct this.
+
+      Fixes GitHub issue #4952
 
 
     Java Library

--- a/src/H5Tinit_float.c
+++ b/src/H5Tinit_float.c
@@ -608,9 +608,13 @@ H5T__init_native_float_types(void)
 #endif
 
 done:
-    /* Clear any FE_INVALID exceptions from NaN handling */
+    /* Clear any FE_INVALID exceptions from NaN handling. FE_INVALID is C99/C11,
+     * but may not be present on all systems.
+     */
+#ifdef FE_INVALID
     if (feclearexcept(FE_INVALID) != 0)
         HSYS_GOTO_ERROR(H5E_DATATYPE, H5E_CANTSET, FAIL, "can't clear floating-point exceptions");
+#endif
 
     /* Restore the original environment */
     if (feupdateenv(&saved_fenv) != 0)

